### PR TITLE
Bring path validation into paths component. Error boundary around sidebar.

### PR DESCRIPTION
**Pull request by sonicyeti**
This moves path validation and array creation out of sidebar root into the paths component itself. This also wraps the sidebar with our error boundary component rounding out or validation approach for sidebar components.

If a component (i.e. servers, paths, requests, etc.) is entirely missing, it is not rendered.
If a component named entry is present but its properties are invalid or missing, we show a warning contextually.
If the spec itself is invalid, we catch and show our error boundary > messaging.

In all scenarios, once fixed, the sidebar re-renders.

All cases can be scene here:

### CODE REVIEW

**Comment by develohpanda** [comment]
I love the error placeholder 😍 

image

File: packages/insomnia-app/app/ui/components/wrapper-design.js [hunk 1]
```diff
@@ -223,10 +223,12 @@ class WrapperDesign extends React.PureComponent<Props, State> {
               <ErrorBoundary
                 invalidationKey={activeApiSpec.contents}
                 renderError={() => (
-                  <div className="text-center margin">
+                  <div className="text-left margin pad">
```
Comment (develohpanda) [inline packages/insomnia-app/app/ui/components/wrapper-design.js:226]: 👍

File: packages/insomnia-components/components/sidebar/index.js [hunk 2]
```diff
@@ -163,9 +159,7 @@ function Sidebar(props: Props) {
         </StyledSection>
       )}
       {serversVisible && servers && <SidebarServers servers={servers} onClick={props.onClick} />}
-      {pathsVisible && pathItems && pathItems.length && (
-        <SidebarPaths paths={pathItems} onClick={props.onClick} />
-      )}
+      {pathsVisible && paths && <SidebarPaths paths={paths} onClick={props.onClick} />}
```
Comment (develohpanda) [inline packages/insomnia-components/components/sidebar/index.js:162]: 👍 Cool, this feels cleaner/more consistent with the other items.

**Review by develohpanda — APPROVED** [review]
LGTM! 💯

**Review by DMarby — APPROVED** [review]

**Pull request merged**